py/server/deephaven/ugp.py

```diff
@@ -122,3 +122,18 @@ def do_locked(*args, **kwargs):
             return f(*args, **kwargs)
 
     return do_locked
+
+
+@contextlib.contextmanager
+def auto_locking_ctx(*args, **kwargs):
+    """A auto-locking aware context manager. It ensures that the enclosed code block runs under
+    the UGP shared lock when ugp.auto_locking is True, the target table-like object is refreshing, and the current
+    thread doesn't own any UGP locks."""
```

Comment:
I think this is missing some information. Same applies to the decorator above, except for the first sentence.
Maybe this is a better wording:
An auto-locking aware context manager. Ensures that the enclosed code block runs under the UGP shared lock if ugp.auto_locking is True, the target table-like object or any table-like arguments are refreshing, and the current thread doesn't own any UGP locks.